src/components/powerInfoPage/chart.tsx

```diff
@@ -0,0 +1,1784 @@
+import React, { useRef, useEffect, useState } from 'react';
+import * as d3 from 'd3';
+import dayjs from 'dayjs';
+import classNames from 'classnames';
+
+interface IApiData {
+  Consume: number;
+  Date: string;
+  Demand: number;
+  ESS: number;
+  EV: number;
+  Generate: number;
+  PV: number;
+  WT: number;
+}
+
+interface IProps {
+  mode: string;
+  date: Date;
+}
+
+interface IPadding {
+  top: number;
+  bottom: number;
+  left: number;
+  right: number;
+  axisX: number;
+}
+
+const Chart: React.FC<IProps> = ({ mode, date }) => {
+  const chartContainer = useRef(null);
+
+  // check whether lastDate is after today or not
+  const lastDate = date.getTime() > new Date().getTime() ? new Date() : date;
+
+  // time format
+  const timeFormat = d3.timeFormat('%Y/%m/%d');
+
+  // axis time format of different width
+  const [axisTimeFormat, setAxisTimeFormat] = useState('%Y/%m/%d');
+
+  // tick padding of axes
+  const [tickPaddingX, setTickPaddingX] = useState(0);
+  const [tickPaddingY, setTickPaddingY] = useState(0);
+
+  // bisector
+  const bisectDate = d3.bisector((d: IApiData) => d.Date).left;
+
+  // calculate first of the chart
+  const firstDate = new Date(lastDate.getTime() - 6 * 24 * 60 * 60 * 1000);
```

Comment:
It's better to use `dayjs.subtract()` to do operation on date.

Reply:
Done. Thanks!